### ISSUE TITLE
obsolete OCCURS not reported

### ISSUE THREAD

**Issue by larshp** [abaplint/abaplint#1839]
```abap
DATA: BEGIN OF li_order OCCURS 0,
          foo TYPE i,
        END OF li_order.
```

should be reported by https://rules.abaplint.org/obsolete_statement/

make sure to check if there is a similar bug when using TYPES

https://github.com/abaplint/abaplint/blob/main/packages/core/src/rules/obsolete_statement.ts

plus add `extendedInformation` regarding OCCURS so it works in the site search
